### PR TITLE
feat: integrate config v2 and implement simple tx backbone

### PR DESCRIPTION
**Pull request by jjeffcaii**
**What this PR does**:
1. Integrate config v2
2. Simple transaction implementation
3. Optimize configuration struct, add tenants

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/executor/redirect.go [hunk 1]
```diff
@@ -181,94 +130,122 @@ func (executor *RedirectExecutor) ExecutorComQuery(ctx *proto.Context) (proto.Re
...
-		case proto.ReadWriteSplitting:
-			return executor.slaveComQueryExecute(ctx, query)
+		if tx, ok := executor.getTx(ctx); ok {
+			res, warn, err = tx.Rollback(ctx.Context)
+		} else {
+			res, warn, err = nil, 0, errors.New("no transaction")
 		}
+	case *ast.SelectStmt:
+		res, warn, err = rt.Execute(ctx)
 	default:
```
Comment (dk-lockdown) [inline pkg/executor/redirect.go:167]: 如果 select 在事务中的话，要用 tx 来执行。增删改同理。

File: pkg/runtime/runtime.go [hunk 2]
```diff
@@ -107,7 +324,58 @@ type myDB struct {
...
+
+	if bc, err = db.borrowConnection(ctx); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	db.pendingRequests.Inc()
+
+	if _, _, err = bc.ExecuteWithWarningCount("begin", true); err != nil {
+		// cleanup if failed to begin tx
+		db.pendingRequests.Inc()
```
Comment (jjeffcaii) [inline pkg/runtime/runtime.go:345]: My mistake, should be `Dec`, and will be fixed soon.

File: pkg/executor/redirect.go (hunk 1 above)
Comment (jjeffcaii) [reply pkg/executor/redirect.go:182]: fixed

**Comment by jjeffcaii** [comment]
How to start?
1. delete `./docker/data`
2. make `docker-build`
3.  change docker-compose.yml:  - "13306:13306" -> "13307:13306"
4. run cmd main
image

**Comment by jjeffcaii** [comment]
How to connect?
1. mycli: `mycli -h127.0.0.1 -P13306 -udksl employees -p123456`
2. run: test/integration_test.go#TestBasicTx TestSimpleSharding

**Review by AlexStocks — APPROVED** [review]

**Review by dongzl — APPROVED** [review]

**Review by Lvnszn — APPROVED** [review]

**Review by Penglq — APPROVED** [review]
LGTM

**Comment by dongzl** [comment]
`start -c ./docker/conf/config.yaml`

**Pull request merged**